File: src/main/java/ch/junamin/domain/ISolvableBoard.java
package ch.junamin.domain;

public interface ISolvableBoard {
    void resetBoard();

    boolean canPlaceShip(int row, int col, int size, char direction);

    void placeShip(int row, int col, int size, char direction);

    void removeShip(int row, int col, int size, char direction);

    boolean isValidSolution();

    int getSize();

    String asText();
}

File: dotnet-core/BimaruApi/Domain/ISolvableBoard.cs
namespace BimaruApi.Domain
{
    public interface ISolvableBoard
    {
        void ResetBoard();
        bool CanPlaceShip(int row, int col, int size, char direction);
        void PlaceShip(int row, int col, int size, char direction);
        void RemoveShip(int row, int col, int size, char direction);
        bool IsValidSolution();
        int GetSize();
        string AsText { get; }
    }
}
